tmt.spec

```diff
@@ -1,234 +1,149 @@
-BuildRequires: python%{python3_pkgversion}-testcloud >= 0.9.10
-BuildRequires: python%{python3_pkgversion}-markdown
-BuildRequires: python%{python3_pkgversion}-junit_xml
-BuildRequires: python%{python3_pkgversion}-ruamel-yaml
-BuildRequires: python%{python3_pkgversion}-jinja2
-BuildRequires: python%{python3_pkgversion}-pint
-# TypeAlias is not available with python3.9 on RHEL9
-%if 0%{?rhel} == 9
-BuildRequires: python%{python3_pkgversion}-typing-extensions
-%endif
-# Required for tests
-BuildRequires: rsync
-%{?python_provide:%python_provide python%{python3_pkgversion}-%{name}}
 
-%description -n python%{python3_pkgversion}-%{name}
-The tmt Python module and command line tool implement the test
-metadata specification (L1 and L2) and allows easy test execution.
-This package contains the Python 3 module.
-
-%package provision-container
-Summary: Container provisioner for the Test Management Tool
-Obsoletes: tmt-container < 0.17
-Requires: tmt == %{version}-%{release}
-Requires: podman
-Requires: (ansible or ansible-collection-containers-podman)
-
-%description provision-container
-Dependencies required to run tests in a container environment.
-
-%package provision-virtual
-Summary: Virtual machine provisioner for the Test Management Tool
-Obsoletes: tmt-testcloud < 0.17
-Requires: tmt == %{version}-%{release}
-Requires: python%{python3_pkgversion}-testcloud >= 0.9.2
-Requires: libvirt-daemon-config-network
-Requires: openssh-clients
-Requires: (ansible or ansible-core)
+%pyproject_extras_subpkg -n tmt export-polarion
+%pyproject_extras_subpkg -n tmt report-junit
+%pyproject_extras_subpkg -n tmt report-polarion
+
+%package -n     tmt+test-convert
+Summary:        Dependencies required for tmt test import and export
+Obsoletes:      tmt-test-convert < 1.28
+Requires:       tmt == %{version}-%{release}
+Requires:       make
+Requires:       python3-bugzilla
+Requires:       python3-nitrate
+Requires:       python3-html2text
+Requires:       python3-markdown
+
+%description -n tmt+test-convert
+This is a metapackage bringing in extra dependencies for tmt.
+It contains no code, just makes sure the dependencies are installed.
+
+%package -n     tmt+provision-container
+Summary:        Dependencies required for tmt container provisioner
+Obsoletes:      tmt-provision-container < 1.28
+Obsoletes:      tmt-container < 0.17
+Requires:       tmt == %{version}-%{release}
+Requires:       podman
+Requires:       (ansible or ansible-collection-containers-podman)
+
+%description -n tmt+provision-container
+This is a metapackage bringing in extra dependencies for tmt.
+It contains no code, just makes sure the dependencies are installed.
+
+%package -n     tmt+provision-virtual
+Summary:        Dependencies required for tmt virtual machine provisioner
+Obsoletes:      tmt-provision-virtual < 1.28
+Obsoletes:      tmt-testcloud < 0.17
+Requires:       tmt == %{version}-%{release}
+Requires:       python3-testcloud
```

Comment:
Shouldn't we specify version here as well?

Reply:
hmm, I think that would mean either specifying minimal version which is in epel9, or doing `if` and check for OS. Same for other rpm dependencies.

Reply:
Yes, the thing is that we depend on recent `testcloud` changes:

https://github.com/teemtee/tmt/blob/c0649ee8b304915fd54934b01fd7dfae0466cc60/setup.py#L78

And it seems we've forgotten to update the specfile as well:

https://github.com/teemtee/tmt/blob/c0649ee8b304915fd54934b01fd7dfae0466cc60/tmt.spec#L74

We should require `>=0.9.10`.